src/cpp/preprocessor/aie2/aie2_blob_preprocessor_input.cpp

```diff
@@ -421,6 +429,16 @@ add_preemption_code(uint32_t col)
           ptr += bw_header->Size;
           break;
         }
+        case XAIE_IO_LOADPDI: {
+          auto lp_header = reinterpret_cast<const XAie_LoadPdiHdr *>(ptr);
+          std::string pdiname = ".pdi."+std::to_string(lp_header->PdiId);
```

Comment:
Please move the PDI name schema logic to a separate function like `std::string get_pdi_name(const XAie_LoadPdiHdr *hdr)
`